libs/store/src/lib/infrastructure/persistence/domain/state-handler.ts

```diff
@@ -0,0 +1,51 @@
+import 'reflect-metadata';
+import { BaseValue } from '../../persistence/domain/base-classes/base-value';
+import { Type } from '../../../domain/public_api';
+
+export function instanceForType<T>(Type: Type<T>, fromState = {}) {
+    const newState = new Type();
+
+    return new Proxy(newState, new StateHandler(fromState));
+}
+/**
+ *  Proxy wrapper
+ */
+class StateHandler<TModel> {
+    constructor(
+        private targetState?: TModel
+    ) {
+    }
+
+    get(target: any, prop: PropertyKey): any {
+        // Check is an array and wraps a new one proxy
+        if (['[object Object]', '[object Array]']
+            .indexOf(Object.prototype.toString.call(this.targetState[prop])) > -1) {
+            return new Proxy(this.targetState[prop], this);
```

Comment:
That means that every read operation will create a new instance of proxy, Am I right?
For example:
```
read entityProxy.lineItems ---> create new instance of Proxy
...
read entityProxy.lineItems ---> create new instance of Proxy 
```

I wonder if we could create only one Proxy instance for each property that needs to be wrapped. Can we do that?

Reply:
Good point. I will check what we can do with it